### PR TITLE
TRUNK-5484: Adding catch expectedException to the class UserService, …

### PR DESCRIPTION
**Pull request by Wandji69**
…to throw MessageException when no email is provided.

https://issues.openmrs.org/browse/TRUNK-5484

<b>Checklist: I completed these to help reviewers :)</b>
- [x] My pull request only contains ONE single 
(the number above, next to the 'Commits' tab is 1).

No? -> <a href="https://wiki.openmrs.org/display/docs/Pull+Request+Tip">read here</a> on how to squash multiple commits into one

 - [x] My IDE is configured to follow the<a href="https://wiki.openmrs.org/display/docs/Java+Conventions"> code style</a> of this project.

No? Unsure? -> <a href="https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE">configure your IDE</a>, format the code and add the changes with `git add . && git commit --amend`

 - [x] I have added tests to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
added all formatting changes to my commit.

No? -> execute above command

 - [x] All new and existing tests passed.

No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

 - [x] My pull request is based on the latest changes of the master branch.

No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/UserServiceTest.java [hunk 1]
```diff
@@ -1403,7 +1403,7 @@ public void setUserActivationKey_shouldCreateUserActivationKey() throws MessageE
 		    "http://localhost:8080/openmrs/admin/users/changePassword.form/{activationKey}");
 		User createdUser = userService.createUser(u, "Openmr5xy");
 		assertNull(dao.getLoginCredential(createdUser).getActivationKey());
-		assertEquals(createdUser, userService.setUserActivationKey(createdUser));
+		expectedException.expect(MessageException.class);
 		assertNotNull(dao.getLoginCredential(createdUser).getActivationKey());
```
[the remark this answers is not in the document]
Comment (Wandji69) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java]: Yes i did run the test and made sure it pass let me check

**Comment by suthagar23** [comment]
@Wandji69 Can you please modify the PR according to the preferred PR template?

File: api/src/test/java/org/openmrs/api/UserServiceTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wandji69) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java]: All test have passed and with no conflict.

**Comment by fanyui** [comment]
@Wandji69  Can you fix the pull request as suggested by suthagar23, Use the template that you were presented when creating this pull request.

**Comment by Wandji69** [comment]
> @Wandji69 Can you please modify the PR according to the preferred PR template?

**Comment by fanyui** [comment]
@Wandji69  Great improvement, Just a small fix now and this will be ready. - [x] instead of just the [x] which you are using. so markdown can interpret as a checked box

**Comment by Wandji69** [comment]
I have done that. Now it's up and Okay!

**Comment by dkayiwa** [comment]
@fanyui do you have any comments for this pull request?

**Comment by fanyui** [comment]
@dkayiwa the pull request is Ok. I have checked it out tested it and it good. Thanks for the fix @Wandji69

**Pull request merged**